rxandroidble/src/main/java/com/polidea/rxandroidble/scan/ScanFilter.java

```diff
@@ -277,8 +277,12 @@ public boolean matches(RxBleInternalScanResult scanResult) {
         }
 
         // Local name match.
-        if (mDeviceName != null && !mDeviceName.equals(scanRecord.getDeviceName())) {
-            return false;
+        if (mDeviceName != null) {
+            final boolean advertisedDeviceNameMatches = mDeviceName.equals(scanRecord.getDeviceName());
+            final boolean bluetoothDeviceNameMatches = mDeviceName.equals(device.getName());
```

Comment:
Small but pricy optimization. You can inline check from line 282 (and extract to a method) to avoid code execution when it's not required.

In this example the second check will be executed regardless the first one (even if deviceName matches).